src/doc/docs/decomposition.md

```diff
@@ -181,41 +179,42 @@ public class ModuleImpl<C extends Configuration> extends DropwizardAwareModule<C
 
 These tips show various techniques for developing bundles.  
 Mostly, these tips are based on developing [guicey extensions](https://github.com/xvik/dropwizard-guicey-ext).
-(look extensions sources as examples).
+See extension sources for examples.
 
 ### Uniqueness
 
-For everything that is registered "by instance" applied [de-duplication mechanism](guide/deduplication.md).
+For everything that is registered "by instance", the [de-duplication mechanism](guide/deduplication.md) is applied.
 
-You can use it to grant only one instance of [bundle](guide/bundles.md#de-duplication) by `extends UniqueGuiceyBundle`
-(or more sophisticated logic by manual [equals and hash code implementation](guide/deduplication.md#equals-method),
-for example, to de-duplicate only instance with the same constructor arguments).
+You can use it to provide only one instance of a [bundle](guide/bundles.md#bundle-de-duplication) by `extends 
+UniqueGuiceyBundle`. If more sophisticated logic is required, a manual [equals and hash code implementation](guide/deduplication.md#equals-method),
+may be used. This could be used to de-duplicate only instances with the same constructor arguments.
 
-It may be mostly important for [guice modules](guide/deduplication.md#guice-modules) as guice will not start with duplicate bundings
+This is most applicable to [guice modules](guide/deduplication.md#guice-modules) as guice will not start with duplicate bindings
 (`MyModule extends UniqueModule`).
 
 ### Auto-loaded bundle
 
-Auto loading based on guicey [bundles lookup](guide/bundles.md#service-loader-lookup).
+Auto-loading is based on guicey [bundles lookup](guide/bundles.md#service-loader-lookup) algorithm.
```

Comment:
I may not know some language nuances, but when I see an "algorithm" I imagine something sophisticated. Bundles lookup implementation is very simple (lookup and load) I don't think it deserves to be called an "algorithm". Probably a "feature" would be better.